src/OrchardCore.Modules/OrchardCore.Users/Localization/Providers/UserLocalizationRequestCultureProvider.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Localization;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Localization;
+
+namespace OrchardCore.Users.Localization.Providers;
+
+public class UserLocalizationRequestCultureProvider : RequestCultureProvider
+{
+    public override async Task<ProviderCultureResult> DetermineProviderCultureResult(HttpContext httpContext)
+    {
+        ArgumentNullException.ThrowIfNull(httpContext);
+
+        if (httpContext?.User?.Identity?.IsAuthenticated == false)
+        {
+            return await NullProviderCultureResult;
```

Comment:
Yes, as said the github search work everywhere unless in the aspnetcore repo for me ;) yes I saw that it's like an helper to use a `Task.FromResult()` when in a method you don't use any `await`.

    NullProviderCultureResult = Task.FromResult(default(ProviderCultureResult));

But when your method already use an `await`, in place of awaiting a `Task.FromResult()`, better to just return `default(ProviderCultureResult)`, so here the type being implied, my suggestion is.

    return default;


 Which returns null because `ProviderCultureResult` is a class (not a value type) so `return null;` was also okay.